src/lin_diffEq_HHL.jl

```diff
@@ -15,13 +17,53 @@ export Array_QuEuler, prepare_init_state, solve_QuEuler
     * h - step size.
     * tspan - time span.
 """
-function prepare_init_state(tspan::Tuple,x::Vector,h::Float64,g::Function)
+
+"""
+    alg_lin_diffeq
+    * step - step for multistep method
+    * α - coefficients for xₙ
+    * β - coefficent for xₙ'
+"""
+struct alg_lin_diffeq
```

Comment:
We mainly follow the [YASGuide](https://github.com/jrevels/YASGuide) for Yao related development. Could you use CamelCase for type names?  and also snake case for method names. I won't comment the rest, there are some similar issues in this PR.